src/plugins/analysis/elf_analysis/code/elf_analysis.py

```diff
@@ -1,161 +1,288 @@
-        return list(set(tags))
+class ElfSection(BaseModel):
+    flags: List[str]
+    name: str
+    size: int
+    type: str
+    offset: int
+    virtual_address: int
 
-    @staticmethod
-    def _get_symbols_version_entries(symbol_versions):
-        imported_libs = []
-        for sv in symbol_versions:
-            if str(sv) != '* Local *' and str(sv) != '* Global *':
-                imported_libs.append(str(sv).split('(', maxsplit=1)[0])
-        return list(set(imported_libs))
+    @classmethod
+    def from_lief_section(cls, section: lief.ELF.Section) -> ElfSection:
+        return cls(
+            flags=[f.__name__ for f in section.flags_list],
+            name=section.name,
+            size=section.size,
+            type=section.type.__name__,
+            offset=section.offset,
+            virtual_address=section.virtual_address,
+        )
 
-    @staticmethod
-    def _get_relevant_imp_functions(imp_functions):
-        imp_functions[:] = [x for x in imp_functions if not x.startswith('__')]
-        return imp_functions
+
+class ElfSegment(BaseModel):
+    file_offset: int
+    flags: List[str]
+    physical_address: int
+    physical_size: int
+    type: str
+    virtual_address: int
+    virtual_size: int
+
+    @classmethod
+    def from_lief_segment(cls, segment: lief.ELF.Segment) -> ElfSegment:
+        return cls(
+            file_offset=segment.file_offset,
+            flags=[ELF_SEGMENT_FLAGS.get(segment.flags.value, 'None')],
+            physical_address=segment.physical_address,
+            physical_size=segment.physical_size,
+            type=segment.type.__name__,
+            virtual_address=segment.virtual_address,
+            virtual_size=segment.virtual_size,
+        )
+
+
+class DynamicEntry(BaseModel):
+    tag: str
+    value: int
+    library: Optional[str] = None
+    flags: Optional[List[str]] = None
+    array: Optional[List[str]] = None
+
+    @classmethod
+    def from_lief_dyn_entry(cls, entry: lief.ELF.DynamicEntry) -> DynamicEntry:
+        return cls(
+            tag=entry.tag.__name__,
+            value=entry.value,
+            library=getattr(entry, 'name', None),
+            flags=[f.__name__ for f in entry.flags] if hasattr(entry, 'flags') else None,
+            array=[str(i) for i in entry.array] if hasattr(entry, 'array') else None,
+        )
+
+
+class ElfSymbol(BaseModel):
+    name: str
+    offset: int
+
+
+class InfoSectionData(BaseModel):
+    name: str
+    contents: str
+
+
+class AnalysisPlugin(AnalysisPluginV0, AnalysisBasePluginAdapterMixin):
+    class Schema(BaseModel):
+        header: ElfHeader
+        sections: List[ElfSection]
+        segments: List[ElfSegment]
+        dynamic_entries: List[DynamicEntry]
+        exported_functions: List[ElfSymbol]
+        imported_functions: List[ElfSymbol]
+        libraries: List[str]
+        mod_info: Optional[List[str]] = Field(description='Key value pairs with module information.')
```

Comment:
Could you reword the description and make more clear that this is related to _Linux kernel_ modules (Isn't it?).
To me the description does not convey much information that is not conveyed by the type and name of the field. 

Also, how do 'key value pairs' fit in an `Optinal[List[str]]` ?

Reply:
I agree, a list of key value pairs doesn't make a lot of sense. I converted it to a dict.